test/write-batch.js

```diff
@@ -368,29 +361,28 @@ describe('batch support', function() {
     // we are running on GCF.
     process.env.FUNCTION_TRIGGER_TYPE = 'http-trigger';
 
-    return createInstance().then(firestore => {
-      firestore._preferTransactions = true;
-      firestore._lastSuccessfulRequest = null;
-
-      let beginCalled = 0;
-      let commitCalled = 0;
+    let beginCalled = 0;
+    let commitCalled = 0;
 
-      firestore._firestoreClient._innerApiCalls.beginTransaction = function(
-          actual, options, callback) {
+    const overrides = {
+      beginTransaction: (actual, options, callback) => {
         ++beginCalled;
         callback(null, {transaction: 'foo'});
-      };
-
-      firestore._firestoreClient._innerApiCalls.commit = function(
-          actual, options, callback) {
+      },
+      commit: (actual, options, callback) => {
         ++commitCalled;
         callback(null, {
           commitTime: {
             nanos: 0,
             seconds: 0,
           },
-        });
-      };
+        })
+      }
+    };
+
+    return createInstance(overrides).then(firestore => {
+      firestore._preferTransactions = true;
```

Comment:
Why is it necessary to set these flags now?

Reply:
`firestore` didn't exist beforehand.